test/common_test/free_joint_features.cc

```diff
@@ -316,6 +316,71 @@ TEST_F(FreeGroupFeaturesTest, FreeGroupSetWorldPosePrincipalAxesOffset)
   }
 }
 
+TEST_F(FreeGroupFeaturesTest, FreeGroupSetWorldPoseStaticModel)
```

Comment:
Do we need to check how it behaves with non-static models attached to the world with a fixed joint?

Reply:
updated code to make sure non-static model with a world joint can not be moved, also added a test a7e6eb5